File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int val=0;
        int min=prices[0];
        for(int i=0;i<prices.length;i++){
                if(prices[i] < min){
                    min=prices[i];
                }
                else if (prices[i] - min > val) {
                val = prices[i] - min;    
                }
            }
        return val;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices){
        int minPrice=INT_MAX,maxProfit=0;
        for(int i=0;i<prices.size();i++){
            if(prices[i]<minPrice)minPrice=prices[i];
            else if(prices[i]-minPrice>maxProfit)maxProfit=prices[i]-minPrice;
        }
        return maxProfit;
    }
};
